flink-core/src/main/java/org/apache/flink/api/common/distributions/DataDistribution.java

```diff
@@ -47,6 +49,12 @@
 	 */
 	Object[] getBucketBoundary(int bucketNum, int totalNumBuckets);
 	
+	public int[] getBoundaryKeyPositions();
```

Comment:
Why add these interfaces? the partition key position is defined by user through `partitionByRange` parameter, and the partition key type information is extracted by flink itself, you should not depends on `DataDistribution` to have these information.